### PR TITLE
Refactoring of the plugin responsible for the image/directory delete functionality + adjust tests

### PR DESCRIPTION
**Pull request by coderimus**
### Description (*)
While working on one of the ASI issues I found one bug and the way we can improve some part of the MediaGallery. Also, fix a bug with deleting an image uploaded to the root media folder.

This PR provides:

* Fix bug for media file uploaded to the media root folder.
* Refactoring to the MediaGallery plugin responsible for image or folder delete functionality. I removed deprecated classes and start to use recommended.
* Adjust Unit tests for the current MediaGallery functionality. They were good but I have to combine them and add changes required to the current implementation.
* Fix _fixture classes for MediaGallery. 

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
N/A

### Manual testing scenarios (*)

**Image delete test**
Magento 2 v2.4-develop
1. Log in to the Magento Admin
2. Go to the `Catalog` -> `Categories`
3. Click on the `Content` collapsable panel and press `Select from gallery`
4. Upload an image and validate that it was uploaded to the physical storage and entry created in `media_gallery_asset` for the uploaded image
5. Select uploaded image in Media gallery and press `Delete Selected` in the Meda gallery control panel

**Accepted result:** image deleted from the physical storage and from the `media_gallery_asset`

**Folder with images delete test**
Magento 2 v2.4-develop
1. Log in to the Magento Admin
2. Go to the `Catalog` -> `Categories`
3. Click on the `Content` collapsable panel and press `Select from gallery`
4. Click on `Create Folder` and create a custom folder
5. Select the newly created folder
4. Upload several images to it and validate that they uploaded to the physical storage and entries created in `media_gallery_asset` for the uploaded images
5. Select created folder with images in Media gallery and press `Delete Folder` in the Meda gallery control panel

**Accepted result:** images deleted from physical storage, folder deleted from physical storage and from the `media_gallery_asset`

### Questions or comments
N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7489](https://jira.corp.magento.com/browse/ENGCOM-7489) has been created to process this Pull Request
:eight_spoked_asterisk: @sivaschenko, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by sivaschenko** [comment]
@coderimus can you please add the testing scenario for QA to be able to check the changes

**Comment by coderimus** [comment]
@sivaschenko QA steps added. Please, validate.

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
:x: QA Failed

The Deleted image reappears in Megia Gallery

**Comment by sivaschenko** [comment]
The issue with deleted image reappearing should be addressed by https://github.com/magento/adobe-stock-integration/pull/1279

**Comment by coderimus** [comment]
@sivaschenko I provide changes for getting a related path for images uploaded to the root of the media directory. Please, review.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7489](https://jira.corp.magento.com/browse/ENGCOM-7489) has been created to process this Pull Request

File: app/code/Magento/MediaGallery/Plugin/Wysiwyg/Images/Storage.php [hunk 1]
```diff
@@ -120,8 +100,8 @@ public function afterDeleteDirectory(StorageSubject $subject, $result, $path)
         }
 
         try {
-            $this->deleteMediaAssetByDirectoryPath->execute($this->getMediaDirectoryRelativePath($path));
-        } catch (ValidatorException $exception) {
+            $this->deleteMediaAssetByPath->execute([$this->getMediaDirectoryRelativePath($path)]);
```
Comment (sivaschenko) [inline app/code/Magento/MediaGallery/Plugin/Wysiwyg/Images/Storage.php]:
```suggestion
            $this->deleteMediaAssetByPath->execute([$this->filesystem->getDirectoryRead(DirectoryList::MEDIA)->getRelativePath($path)]);
```

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for updates @coderimus ! I don't think the path corrections should be applied when the directory is deleted. Can you please take a look at my correction suggestion and check this case

**Comment by coderimus** [comment]
@sivaschenko requested changes added. Please, review.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7489](https://jira.corp.magento.com/browse/ENGCOM-7489) has been created to process this Pull Request

**Comment by engcom-Golf** [comment]
:heavy_check_mark: QA Passed  images deleted from physical storage, folder deleted from physical storage and from the media_gallery_asset
**Note** Tested with linked PR on AdobeStock repository https://github.com/magento/adobe-stock-integration/pull/1279

**Review by sivaschenko — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-7489](https://jira.corp.magento.com/browse/ENGCOM-7489) has been created to process this Pull Request

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**